### PR TITLE
feat: partial GWT support for network status service

### PR DESCRIPTION
**Pull request by marcellorinaldo**
Signed-off-by: Marcello Martina <martina.marcello.rinaldo@outlook.com>

This PR adds a partial support to the GWT component to retrieve status information.

**Related Issue:** N/A.

**Description of the solution adopted:** This PR adds the implementation for:
- Common general status information
- DHCP client address
- DNS servers address list
- Partial modem state properties

**Screenshots:** N/A.

**Any side note on the changes made:** N/A.

### CODE REVIEW

**Comment by marcellorinaldo** [comment]
TODO:
- [x] Read interface list only from status
- [x] If interface is in status but not in config, then put it as disabled
- [x] Displayed Hw State is not correct

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/status/NetworkStatusServiceAdapter.java [hunk 1]
```diff
@@ -31,57 +39,106 @@ public class NetworkStatusServiceAdapter {
...
-        this.gwtConfig.setHwRssi(NA);
+    @SuppressWarnings("restriction")
+    private void setCommonStateProperties(GwtNetInterfaceConfig gwtConfig,
+            NetInterface<NetInterfaceAddress> networkInterface) {
+
+        if (networkInterface.getState() != null) {
+            gwtConfig.setHwState(networkInterface.getState().name());
+        }
+        if (networkInterface.getType() != null) {
+            gwtConfig.setHwState(networkInterface.getType().name());
```
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/status/NetworkStatusServiceAdapter.java]: Should be `type` here

**Review by mattdibi — APPROVED** [review]
Tested working on [mattdibi:matt/experimental/nm_dbus_integration](https://github.com/mattdibi/kura/tree/matt/experimental/nm_dbus_integration). The Ethernet interface informations are correctly retrieved and displayed by the WebUI

image
image
image

**Comment by mattdibi** [comment]
Working correctly for loopback interface too:

image

**Pull request merged**